### PR TITLE
CI: Switch from Travis to Github Actions.

### PR DESCRIPTION
**Pull request by cpu**
### Description 

Travis CI has changed their product in such a way that open source projects are seeing long queue times. Since we're already using Github it makes sense to use Github actions for CI. This removes an external dependency and seems like a more modern approach overall. Also it turns out overall CI times are also better with Github actions, a nice bonus. Similar work was done for [ZCrypto](https://github.com/zmap/zcrypto/pull/230) already.

New workflows:
* `.github/workflows/go.yml` - handles building + unit tests.
* `.github/workflows/golangci-lint.yml` - handles linting Go source code (notably also handles the `gofmt` check).
* `.github/workflows/testdata-lint.yml` - handles linting unit test .pem files to check for OpenSSL textual representation before header.
* `.github/workflows/integration.yml` - handles cert corpus caching and integration tests.
* `.github/workflows/release.yml` - handles invoking Goreleaser on release tag pushes.

Notable changes:

* Golanci-lint had to be updated to a newer version to support the action. I fixed a few of the new linter findings and added ignores for others.

* Integration tests - The Github Action cache support is much more generous (5GB!). This lets us cache the integration test corpus data between runs saving ~1-2m each run.

* Cron schedule - I upped the jobs to run automatically (in addition to PRs/branch pushes) every 12 hour where prev. they ran every 24hr.

Resolves https://github.com/zmap/zlint/issues/500

### Testing

To check that all of these new checks do what they're supposed to I pushed [a branch](https://github.com/cpu/zlint/tree/cpu-test-gh-actions-enforce) that has commits intended to break each stage:

* Unit test failure - https://github.com/cpu/zlint/commit/c0e88c912dcaa1b8037ac333a006afdcab18a88d and [associated action failure](https://github.com/cpu/zlint/runs/1467973114?check_suite_focus=true)
* Linter failure - https://github.com/cpu/zlint/commit/48020bfea8445330500129fcb7e599586e665a9e and [associated action failure](https://github.com/cpu/zlint/actions/runs/388955388)
* Testdata missing openssl prepend - https://github.com/cpu/zlint/commit/34cf3d09e2c1ceede00229e0a3387f5097ead579 and [associated action failure](https://github.com/cpu/zlint/runs/1467973110?check_suite_focus=true)
* Integration test failure - https://github.com/cpu/zlint/commit/bf7316b940d2e2371756944915aa71d09c3500cf and [associated action failure](https://github.com/cpu/zlint/runs/1467973113?check_suite_focus=true)

### Follow-up

* We'll need to disable the existing master Travis branch protections and enable branch protections for the Github Actions workflows.
* It would be nice to rework the gtld update PR automation to use a workflow defined in this repo. - will file a follow-up issue for that.

### CODE REVIEW

File: v2/cmd/zlint-gtld-update/main.go [hunk 1]
```diff
@@ -108,7 +109,16 @@ var tldMap = map[string]GTLDPeriod{
 // getData fetches the response body bytes from an HTTP get to the provider url,
 // or returns an error.
 func getData(url string) ([]byte, error) {
-	resp, err := httpClient.Get(url)
```
Comment (cpu) [inline v2/cmd/zlint-gtld-update/main.go:111]: This was flagged by the `noctx` linter so I quickly reworked it to use a timeout.

File: v2/lints/rfc/lint_ca_subject_field_empty.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func (l *caSubjectEmpty) CheckApplies(c *x509.Certificate) bool {
 }
 
 func (l *caSubjectEmpty) Execute(c *x509.Certificate) *lint.LintResult {
-	if &c.Subject != nil && util.NotAllNameFieldsAreEmpty(&c.Subject) {
```
Comment (cpu) [inline v2/lints/rfc/lint_ca_subject_field_empty.go:46]: pointer to a variable addr can never be nil.

File: v2/lints/rfc/lint_issuer_field_empty.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ func (l *issuerFieldEmpty) CheckApplies(c *x509.Certificate) bool {
 }
 
 func (l *issuerFieldEmpty) Execute(c *x509.Certificate) *lint.LintResult {
-	if &c.Issuer != nil && util.NotAllNameFieldsAreEmpty(&c.Issuer) {
```
Comment (cpu) [inline v2/lints/rfc/lint_issuer_field_empty.go:42]: pointer to a variable addr can never be nil.

File: v2/util/algorithm_identifier.go [hunk 4]
```diff
@@ -11,13 +11,13 @@ import (
 	cryptobyte_asn1 "golang.org/x/crypto/cryptobyte/asn1"
 )
 
-// additional OIDs not provided by the x509 package
```
Comment (cpu) [inline v2/util/algorithm_identifier.go:14]: I fixed some `godot` nits in this file and then gave up when I saw how many others needed updating. I added `godot` to the linter ignore list for now.

**Comment by cpu** [comment]
You can see the successful workflow executions [here on my fork](https://github.com/cpu/zlint/actions?query=branch%3Acpu-gh-actions). I'm not sure why they aren't running here yet except that this hasn't been merged.

**Review by zakird — APPROVED** [review]

**Comment by cpu** [comment]
I'm going to administratively merge this PR and disable the Travis integration.

**Comment by cpu** [comment]
Everything looks good to me post-merge:

ci actions

ci badges

I was also able to update the branch protections for this repo:

ci protections

**Pull request merged**